core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -418,7 +422,11 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
                         throw new VersionConflictEngineException(shardId, get.type(), get.id(),
                             get.versionType().explainConflictForReads(versionValue.version, get.version()));
                     }
+                    if (onRefresh == null)
+                        throw new InvalidParameterException("Consumer onRefresh should not be null in case of realtime get.");
```

Comment:
This is the wrong exception type (it's a security exception), and I'm not sure if we even need to throw here, just let it NPE?